### ISSUE TITLE
Rebuilding Pytorch after modifying C++ code in aten library

### ISSUE THREAD

**Issue by knandanan** [pytorch/pytorch#7370]
Would python setup.py build be enough to rebuild the pytorch after changing C++/Cuda code in the aten library ? For example  I would like to modify the forward and back propagation for batch normalization and I intend to make changes in BatchNormalization.cu in THC library . How would I go about rebuilding the code. I apologise in advance if the question is really trivial as I am a beginner to PyTorch :)

**Comment by ezyang** [comment]
No, you need to run `build_deps`. When I'm compiling PyTorch from source I tend to run `python setup.py build_deps develop` (make sure you haven't installed it previously; if you have, `pip uninstall torch` twice will fix it.)

See also https://github.com/pytorch/pytorch/blob/master/CONTRIBUTING.md

**Comment by knandanan** [comment]
Thank you so much for the prompt response ! Much appreciated!

**Comment by cdemirsoy** [comment]
In the https://github.com/pytorch/pytorch/blob/master/CONTRIBUTING.md, it was mentioned that running the following command will satisfy the need if the modified files are from aten or torch/csrc. 
            
      -  python setup.py rebuild_libtorch

Isn't this correct? Isn't there any way to quickly compile for the changes from C++ or CUDA files? In our case, "python setup.py rebuild_libtorch" command did not work, and compiling from scratch takes too much time.

Thanks in advance!

**Comment by knandanan** [comment]
Good to know! python setup.py build_deps worked for me ...

**Comment by cdemirsoy** [comment]
Yes, but it takes too much time (i.e., 45 minutes on my google cloud). The command I mentioned takes around a minute and gets the work done.
